File: src/domain/repositories/assessmentRepository.ts
import { AssessmentResult } from '../entities/assessmentResult';
import { AssessmentScale } from '../entities/assessmentScale';
import { Patient } from '../entities/patient';

export interface AssessmentRepository {
  getPatients(): Promise<Patient[]>;
  getScales(): Promise<AssessmentScale[]>;
  getRecentResults(): Promise<AssessmentResult[]>;
  submitAssessment(
    patientId: string,
    scaleId: string,
    responses: Record<string, number | boolean | string>
  ): Promise<AssessmentResult>;
}

File: lib/domain/repositories/assessment_repository.dart
import '../entities/assessment_result.dart';
import '../entities/assessment_scale.dart';
import '../entities/patient.dart';

typedef DashboardSnapshot = Map<String, double>;

typedef TrendPoint = ({String label, double value});

typedef InsightCard = ({String title, String subtitle, double change});

typedef AssessmentFlow = ({
  AssessmentScale scale,
  List<ScaleQuestion> flattenedQuestions,
});

abstract class AssessmentRepository {
  Future<DashboardSnapshot> getDashboardSnapshot();

  Future<List<Patient>> getPatients();

  Future<List<AssessmentResult>> getRecentResults();

  Future<List<AssessmentScale>> getAssessmentScales();

  Future<AssessmentResult> submitAssessment({
    required String patientId,
    required String scaleId,
    required Map<String, int> answers,
  });

  Future<List<TrendPoint>> getProgressTrend(String patientId);

  Future<List<InsightCard>> getInsightCards();
}
